gossipd/routing.c

```diff
@@ -877,17 +877,39 @@ static void dijkstra(struct routing_state *rstate,
 		     u64 riskbias,
 		     double fuzz, const struct siphash_seed *base_seed,
 		     struct unvisited *unvisited,
+		     size_t max_nodes,
+		     bool *nodes_limit,
 		     costfn_t *costfn)
 {
 	struct node *cur;
+	size_t scan_nodes = 0;
 
+	if (!max_nodes) {
+		*nodes_limit = true;
+		return;
+	}
+
+	*nodes_limit = false;
 	while ((cur = first_unvisited(unvisited)) != NULL) {
 		update_unvisited_neighbors(rstate, cur, me,
 					   riskfactor, riskbias,
 					   fuzz, base_seed, unvisited, costfn);
+
+		if (*nodes_limit)
+			return;
+
 		remove_unvisited(cur, unvisited, costfn);
-		if (cur == dst)
+
+		if (cur == dst) {
+			status_info("dijkstra: scan %zu nodes", scan_nodes);
```

Comment:
Probably needs to go, or at least be a debug!

Reply:
Fixed!